cmd/ocm/edit/machinepool/cmd.go

```diff
@@ -146,7 +173,52 @@ func run(cmd *cobra.Command, argv []string) error {
 		machinePoolBuilder = machinePoolBuilder.Taints(taintBuilders...)
 	}
 
-	if cmd.Flags().Changed("replicas") {
+	isMinReplicasSet := cmd.Flags().Changed("min-replicas")
+	isMaxReplicasSet := cmd.Flags().Changed("max-replicas")
+	isReplicasSet := cmd.Flags().Changed("replicas")
+
+	// Inferr autoscaling from flags - to avoid getting the existing machine pool
+	// defering to OCM validators
+	if args.autoscalingEnabled || isMaxReplicasSet || isMinReplicasSet {
+		args.autoscalingEnabled = true
+
+		if isReplicasSet {
+			return fmt.Errorf("--replicas can't be set with autoscaling paramteres")
+		}
+	}
+
+	// Editing the default machine pool is a different process
+	if machinePoolID == "default" {
+		clusterConfig := c.Spec{
+			Autoscaling: c.Autoscaling{
+				Enabled:     args.autoscalingEnabled,
+				MinReplicas: args.minReplicas,
```

Comment:
please add validation checks for min/max replicas values

Reply:
ocm cli edit didn't validate any of the edit values before this change. It relies on clusters service validations.

Currently ocm cli only validates the values in cluster create.

That is because validation is more complex and requires getting the cluster and the machine pool.

Do we want to validate these values in ocm-cli?

Reply:
@igoihman I added validations for the default machine pool.

Before the validations this is the error the user got:
```
Error: Failed to update machine pool 'default' on cluster 'zg-j31': identifier is '400', code is 'CLUSTERS-MGMT-400' and operation identifier is '1iim11mj6epg0lu178m4tin487rhu91d': Invalid number of compute nodes: 1. Minimal requirement is 2 compute nodes.
```

This is the error after the validation:
```
Error: Minimum is 2 nodes

```

I am not sure if the validation is required.
It might even be confusing for the user that only wants to update max-replicas on a machine pool that has autoscaling 4-6 nodes.
If the user tries to set max-replicas=1 they will get an error that the minimum is 2 (but the minimum in this case would be 4 - the num of min-replicas).

Since it's a complex validation to check what changed, in CS we first update the DB - validate the latest values, and if an error occurs we rollback - which obviously we can't do in the CLI.

I didn't validate for the additional machine pool - which is even more complex

Thinking again I think I should remove the max-replicas validation I just added. WDYT?